src/main/org/firebirdsql/gds/ng/FbDatabase.java

```diff
@@ -329,4 +326,39 @@
      * @return A potentially cached empty row descriptor for this database.
      */
     RowDescriptor emptyRowDescriptor();
+
+    /**
+     * Creates a batch.
+     *
+     * @param transaction
+     *         Transaction associated with the batch.
+     * @param statement
+     *         Sql statement for creating batch
+     * @param metadata
+     *         Input metadata {@link  FbMessageMetadata}
+     * @param parameters
+     *         Batch parameters buffer {@link BatchParameterBuffer}
+     * @return Instance of {@link FbBatch}
+     * @throws SQLException
+     */
+    FbBatch createBatch(FbTransaction transaction, String statement, FbMessageMetadata metadata, BatchParameterBuffer parameters) throws SQLException;
```

Comment:
I don't think this belongs in `FbDatabase`, it should be part of `FbStatement` if anything.

However, in its current form, I don't think that these methods should be part of the gds-ng API. I see a lot of mismatches with the requirements for JDBC batch execution and the normal usage of the gds-ng API. 

Making this part of the gds-ng API now, will only lead to a lot of rearchitecturing and breaking API changes later when I will try to make a uniform API that can be used from the JDBC part of the driver.

Reply:
I'll try to think about it. Are there any templates of uniform interface?